File: python_solutions/day1.py
# -*- coding: utf-8 -*-

import sys

data = [arg for arg in sys.argv[1:] if not arg.startswith("-")][0]

### DAY 1

## PART 1

with open(data, "r") as f:
    calories = list()
    elfCalories = 0

    for line in f:
        if line == "\n":
            calories.append(elfCalories)
            elfCalories = 0
        else:
            elfCalories += int(line.strip())

print(f"The elf that is carrying the most calories is carrying {max(calories)} calories.")


## PART 2

calories.sort(reverse=True)

print(f"The top three elves are carrying a total of {sum(calories[0:3])} calories.")

File: R_solutions/day1.R
library(data.table)

### DAY 1

args <- commandArgs(trailingOnly = T)
data <- args[which(args == "-in") + 1]

data <- readLines(data)

## PART 1

elf.separators <-c(0, which(data==""), length(data)+1)

calories.per.elf <- data.table(
  elfID=paste("elf", 1:(length(elf.separators)-1), sep="_"),
  calories=unlist(lapply(1:(length(elf.separators)-1), 
                         function(i) {
                           return(sum(as.numeric(data[(elf.separators[i]+1):(elf.separators[i+1]-1)])))})))

cat("The elf that is carrying the most calories is carrying", 
    max(calories.per.elf$calories),
    "calories.\n",
    sep = " ")


## PART 2

cat("The top three elves are carrying a total of", 
    sum(calories.per.elf[order(calories, decreasing = T)][1:3, calories]),
    "calories.\n",
    sep = " ")
